addon/utils.js

```diff
@@ -160,20 +159,27 @@ export function getMergedConfig (cellConfig, cellDefinitions) {
  * @returns {BunsenCell} merged cell definition
  */
 export function getMergedConfigRecursive (cellConfig, cellDefinitions) {
-  const mergedConfig = getMergedConfig(cellConfig, cellDefinitions)
+  let mergedConfig = getMergedConfig(cellConfig, cellDefinitions)
 
   // recursive object case
   if (mergedConfig.children) {
-    const mergedChildConfigs = []
-    mergedConfig.children.forEach((childConfig) => {
-      mergedChildConfigs.push(getMergedConfigRecursive(childConfig, cellDefinitions))
-    })
-    mergedConfig.children = mergedChildConfigs
+    const children = mergedConfig.children.map((childConfig) => getMergedConfigRecursive(childConfig, cellDefinitions))
+    const childrenChanged = children.some((child, index) => child !== mergedConfig.children[index])
+
+    if (childrenChanged) {
+      mergedConfig = Object.assign({}, mergedConfig, {children})
```

Comment:
This is keeping the function pure. Otherwise it can end up mutating the functions input(s). `mergedConfig` isn't actually a new object when the `cellConfig` passed into `getMergedConfig` doesn't contain the `extends` property.